landscape/client/package/tests/test_changer.py

```diff
@@ -1762,3 +1624,16 @@ def assert_session_id(ignored):
         self.changer._session_id = None
         result = self.changer.run()
         return result.addCallback(assert_session_id)
+
+    def test_reboot_flag(self):
+        self.dbus_mock = patch(
+            "landscape.client.package.changer.dbus").start()
+        bus_object = Mock()
+
+        self.dbus_mock.SystemBus.return_value = bus_object
+
+        self.changer._run_reboot()
+
+        def check(_):
+            bus_object.get_object.assert_called_once()
+            bus_object.Reboot.assert_called_once()
```

Comment:
Still need `check` to be added as a callback to a returned `deferred` so that it is actually executed.